modules/data-entry/src/main/frontend/src/vocabQuery/VocabularyBrowser.jsx

```diff
@@ -0,0 +1,216 @@
+//  or more contributor license agreements.  See the NOTICE file
+//  distributed with this work for additional information
+//  regarding copyright ownership.  The ASF licenses this file
+//  to you under the Apache License, Version 2.0 (the
+//  "License"); you may not use this file except in compliance
+//  with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing,
+//  software distributed under the License is distributed on an
+//  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+//  KIND, either express or implied.  See the License for the
+//  specific language governing permissions and limitations
+//  under the License.
+//
+import classNames from "classnames";
+import React, { useRef, useState, useEffect } from "react";
+import PropTypes from "prop-types";
+import { Snackbar, SnackbarContent } from "@material-ui/core";
+import { withStyles } from "@material-ui/core";
+
+import VocabularyTree from "./VocabularyTree.jsx";
+import InfoBox from "./InfoBox.jsx";
+import { REST_URL, MakeRequest } from "./util.jsx";
+import QueryStyle from "./queryStyle.jsx";
+
+// Component that renders a vocabulary info box and browser.
+//
+// Required arguments:
+// onClose: Callback for the vocabulary browser close event
+// infoPath: Term @path to get the term info
+// infoButtonRefs: References to the term info buttons
+// infoboxRef: Reference to the term info dialog node
+// browserRef: Reference to the vocabulary tree dialog node
+//
+// Optional arguments:
+// browserOpen: Boolean representing whether or not the vocabulary tree dialog is open
+//
+function VocabularyBrowser(props) {
+  const { browserOpen, onClose, infoPath, infoButtonRefs, infoboxRef, browserRef, classes } = props;
+
+  const [termInfoVisible, setTermInfoVisible] = useState(false);
+  const [term, setTerm] = useState({});
+
+  const [browsePath, setBrowsePath] = useState("");
+  const [browserOpened, setBrowserOpened] = useState(!!browserOpen);
+
+  const [vocab, setVocab] = useState({});
+
+  const [snackbarVisible, setSnackbarVisible] = useState(false);
+  const [snackbarMessage, setSnackbarMessage] = useState('');
+
+  // Info box properties
+  const [infoAboveBackground, setInfoAboveBackground] = useState(false);
+  const [infoAnchor, setInfoAnchor] = useState(null);
+
+  const [buttonRefs, setButtonRefs] = useState(infoButtonRefs || {});
+  const [noResults, setNoResults] = useState(false);
+
+  useEffect(() => {
+    infoPath ? getInfo(infoPath) : setTermInfoVisible(false);
+  }, [infoPath])
+
+  // Event handler for clicking away from the info box or browser while they are open
+  let clickAwayInfo = (event) => {
+    if ( browserRef?.current?.contains(event.target)
+         || infoboxRef?.current?.contains(event.target)) {
+      return;
+    }
+
+    setTermInfoVisible(false);
+    setInfoAboveBackground(false);
+    onClose && onClose();
```

Comment:
Both @sashaandjic and I think clicking outside of the infoBox, including anywhere on the browser when the infobox is on top, should hide the infoBox.

Would this work?
```suggestion
  let clickAwayInfo = (event) => {
    if ( !infoAboveBackground && browserRef?.current?.contains(event.target)
         || infoboxRef?.current?.contains(event.target)) {
      return;
    }

    setTermInfoVisible(false);
    setInfoAboveBackground(false);
```